### DISCUSSION TITLE
How to change default browser when type "gx"

### DISCUSSION THREAD

**Discussion by tranbinh1604** [LazyVim/LazyVim#4917] [Q&A] [answered]
Hi,

I'm not sure if what I'm going to ask below belongs to LazyVim or not, but I'm a LazyVim user, and I want to ask.

Normally, when I want to open a URL while using LazyVim, I just type "gx," and it will open the URL using the default browser—mine is Safari. Is there any way to change it to open in Google Chrome?

I tried to:

1. Set the `netrw_browser_viewer`: `vim.g.netrw_browsex_viewer = "open -a 'Google Chrome'"`
2. Create a key mapping to overwrite the default keymap

```
vim.keymap.set("n", "gx", function()
  local url = vim.fn.expand("<cfile>")
  vim.fn.system({"open", "-a", "Google Chrome", url})
end, { silent = true })

```

Any help is appreciated.

**Comment by dpetka2001** [comment, marked as answer]
You can find [here in the Neovim repo](https://github.com/neovim/neovim/blob/ab9cfc4dc3422af5235759efef456d3e02745217/runtime/lua/vim/ui.lua#L136-L172). This is what `gx` calls under the hood. So, depending on your OS it will call the corresponding command which will open the default browser. At the end it just calls `vim.system` with the command. 

So, your example with the keymap could be something like 
```lua
vim.keymap.set("n", "gx", function()
  local url = vim.fn.expand("<cfile>")
  vim.system({ "google-chrome", url }, { detach = true})
end, { silent = true })
```

**Reply by tranbinh1604** [reply]
Thank you @dpetka2001

I'm using macOS and tried your keymap as well, but it does not work.

I even disabled netew by adding `vim.g.netrw_nogx = 1` to `/config/lazy.lua`, but it was of no help.

I also tried to disable this keymap `gx` and then create another key mapping in `keymap.lua`, but that didn't help either.

**Reply by tranbinh1604** [reply]
hi @dpetka2001

After reinstalling LazyVim and removing all `~/.local/{state/nvim, share/nvim}` and `~/.cache/nvim`, it works as expected. Thank you so much!
